apps/openmw/mwsound/soundmanagerimp.cpp

```diff
@@ -1068,8 +1076,15 @@ namespace MWSound
                 ++snditer;
         }
 
-        SaySoundMap::iterator sayiter = mActiveSaySounds.begin();
-        while(sayiter != mActiveSaySounds.end())
+        SaySoundMap::iterator sayiter = mSaySoundsQueue.begin();
+        while (sayiter != mSaySoundsQueue.end())
+        {
+            mActiveSaySounds[sayiter->first] = sayiter->second;
+            mSaySoundsQueue.erase(sayiter++);
+        }
```

Comment:
Looks better, but the placement of this hand-off from the queue to the active list is behind the 30hz update check. That would mean there may or may not be multiple frames after `Say` before `SayDone` starts giving 0. Putting this loop before the `if(timePassed < (1.0f/30.0f))` check may fix that.

Reply:
OK, updated.